src/osd/PrimaryLogPG.cc

```diff
@@ -5750,12 +5751,17 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	    soid, op.extent.offset, op.extent.length, osd_op.indata, op.flags);
 	}
 
-	if (op.extent.offset == 0 && op.extent.length >= oi.size)
+	if (op.extent.offset == 0 && op.extent.length >= oi.size
+            && !skip_data_digest) {
 	  obs.oi.set_data_digest(osd_op.indata.crc32c(-1));
-	else if (op.extent.offset == oi.size && obs.oi.is_data_digest())
-	  obs.oi.set_data_digest(osd_op.indata.crc32c(obs.oi.data_digest));
-	else
+	} else if (op.extent.offset == oi.size && obs.oi.is_data_digest()) {
+          if (skip_data_digest)
+            obs.oi.clear_data_digest();
```

Comment:
if the object never sets the digest, why bother clearing it?

Reply:
I intent to take the chance to cleanup any previous existing BlueStore data_digest since this change is going to land on mimic(post-luminous)?

Reply:
i understand you do this with good intention, but what is the merit of doing so? or put in other words, what the downside if we have an "oi.is_data_digest()" on a store with builtin checksum support? will we run into some trouble under such circumstance?

Reply:
The problem is: if we keep **appending** an existing object  previously with data_digest on(instead of writing new objects or random writing), we'll be busy in generating new data_digests; hence the skip_data_digest has no effect and no benefits will be gained by applying this change

Reply:
> will we run into some trouble under such circumstance?

Not really

Reply:
> if we keep appending an existing object previously with data_digest on(instead of writing new objects or random writing), we'll be busy in generating new data_digests; 

ahh, right! i thought we skipped this if the store has builtin checksum support.